rollup/internal/controller/relayer/l2_relayer.go

```diff
@@ -1120,6 +1186,85 @@ func (r *Layer2Relayer) StopSenders() {
 	}
 }
 
+// fetchBlobFeeHistory returns the last WindowSec seconds of blob‐fee samples,
+// by reading L1Block table’s BlobBaseFee column.
+func (r *Layer2Relayer) fetchBlobFeeHistory(windowSec uint64) ([]*big.Int, error) {
+	// how many blocks ago? ~12s per block
+	blocksAgo := windowSec / 12
+	latest, err := r.l1BlockOrm.GetLatestL1BlockHeight(r.ctx)
+	if err != nil {
+		return nil, fmt.Errorf("GetLatestL1BlockHeight: %w", err)
+	}
+	start := int64(latest) - int64(blocksAgo)
+	if start < 0 {
+		start = 0
+	}
+
+	// pull all L1Blocks in [start .. latest]
+	filters := map[string]interface{}{
+		"number >= ?": start,
+		"number <= ?": latest,
+	}
+	recs, err := r.l1BlockOrm.GetL1Blocks(r.ctx, filters)
+	if err != nil {
+		return nil, fmt.Errorf("GetL1Blocks: %w", err)
+	}
+	hist := make([]*big.Int, len(recs))
+	for i, b := range recs {
+		hist[i] = new(big.Int).SetUint64(b.BlobBaseFee)
+	}
```

Comment:
Better to just query the field (`BlobBaseFee`) and not the whole block.

Also, performance-wise this might be problematic (e.g. query 3600 blocks every couple of seconds). You might need to add a db index. Or better, maintain the fee history (or the moving average) in memory instead of querying blocks again every time.